pkgs/swift2objc/lib/src/parser/parsers/declaration_parsers/parse_initializer_declaration.dart

```diff
@@ -0,0 +1,69 @@
+import '../../../ast/_core/shared/parameter.dart';
+import '../../../ast/declarations/compounds/members/initializer_declaration.dart';
+import '../../_core/json.dart';
+import '../../_core/parsed_symbolgraph.dart';
+import '../../_core/utils.dart';
+
+InitializerDeclaration parseInitializerDeclaration(
+  Json initializerSymbolJson,
+  ParsedSymbolgraph symbolgraph,
+) {
+  return InitializerDeclaration(
+    id: parseSymbolId(initializerSymbolJson),
+    params: _parseInitializerParams(initializerSymbolJson, symbolgraph),
+    hasObjCAnnotation: symbolHasObjcAnnotation(initializerSymbolJson),
+  );
+}
+
+List<Parameter> _parseInitializerParams(
```

Comment:
How similar is this to parsing method params? Can they be factorized at all, or are they stored in a totally different way?

Reply:
In methods there's a `functionSignature` field that represents parameters and return type of a method. Each param has its own sub-json object which makes it easier to iterate over & extract params, but unfortunately this field does not exist with initializers, only the flat `declarationFragments` field.

It's worth noting that method has both `functionSignature` and `declarationFragments` fields, so we can make a unified param parsing method using `declarationFragments`, but `functionSignature` felt more "stable" to me so I decided to keep using it with methods.